eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala

```diff
@@ -43,8 +44,11 @@ case class PaymentRequest(prefix: String, amount: Option[MilliSatoshi], timestam
 
   amount.foreach(a => require(a > 0.msat, s"amount is not valid"))
   require(tags.collect { case _: PaymentRequest.PaymentHash => }.size == 1, "there must be exactly one payment hash tag")
-  require(tags.collect { case _: PaymentRequest.PaymentSecret => }.size <= 1, "there must be at most one payment secret tag")
   require(tags.collect { case PaymentRequest.Description(_) | PaymentRequest.DescriptionHash(_) => }.size == 1, "there must be exactly one description tag or one description hash tag")
+  require(!features.allowMultiPart || features.allowPaymentSecret, "there must be a payment secret when using multi-part")
```

Comment:
I think the following is less error-prone:

```scala
if (features.allowMultiPart) {
  require(features.allowPaymentSecret, "there must be a payment secret when using multi-part")
}
```

Reply:
Agreed, done.